### DISCUSSION TITLE
GDS imports, A tutorial roadmap for metamaterial simulation?

### DISCUSSION THREAD

**Discussion by Srinivas-Ganti** [NanoComp/meep#2126] [Q&A] [answered]
Hello everyone,

I have just come across meep and have managed to get it running on my Linux machine (without parallelization). I am going through the various tutorials on the [docs page](https://meep.readthedocs.io/). I am interested in simulating metamaterials in the THz regime. Any guides or useful pages would be highly appreciated. Also, is there a way to import GDS or STEP files to define geometries in meep? 

Edit: About [GDS import](https://meep.readthedocs.io/en/latest/Python_User_Interface/#gdsii-support), Once the file is loaded how can we view it to be sure it is the right geometry? Is there a viewer for this?

**Comment by oskooi** [comment, marked as answer]
See [FAQ/Does Meep support importing GDSII files?](https://meep.readthedocs.io/en/latest/FAQ/#does-meep-support-importing-gdsii-files)

There are three different ways to visualize the geometry in Meep: (1) 2D cross sections using [plot2D](https://meep.readthedocs.io/en/latest/Python_User_Interface/#data-visualization), (2) obtain the ε data in memory as a NumPy array using [`Simulation.get_epsilon()`](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#visualizing-3d-structures) and visualize it using e.g. matplotlib or mayavi, and (3) output the ε data to an HDF5 file on disk and visualize in post processing. The tutorials provide examples of all of these visualization methods.

Note that (1) tends to be the fastest method because it does not involve initializing the structure and fields classes internally. Also note that the ε data may contain *intermediate* values because of the [interpolation from the Yee grid](https://meep.readthedocs.io/en/latest/FAQ/#why-doesnt-turning-off-subpixel-averaging-work).
